rllib/env/single_agent_env_runner.py

```diff
@@ -555,33 +548,72 @@ def _sample_episodes(
 
         return samples
 
-    # TODO (sven): Remove the requirement for EnvRunners to have this
-    #  API. Instead Algorithm should compile episode metrics itself via its local
-    #  buffer.
-    def get_metrics(self) -> List[RolloutMetrics]:
+    def get_metrics(self) -> ResultDict:
         # Compute per-episode metrics (only on already completed episodes).
-        metrics = []
         for eps in self._done_episodes_for_metrics:
             assert eps.is_done
             episode_length = len(eps)
-            episode_reward = eps.get_return()
+            episode_return = eps.get_return()
+            episode_duration_s = eps.get_duration_s()
             # Don't forget about the already returned chunks of this episode.
             if eps.id_ in self._ongoing_episodes_for_metrics:
                 for eps2 in self._ongoing_episodes_for_metrics[eps.id_]:
                     episode_length += len(eps2)
-                    episode_reward += eps2.get_return()
+                    episode_return += eps2.get_return()
+                    episode_duration_s += eps2.get_duration_s()
                 del self._ongoing_episodes_for_metrics[eps.id_]
 
-            metrics.append(
-                RolloutMetrics(
-                    episode_length=episode_length,
-                    episode_reward=episode_reward,
-                )
+            # Log general episode metrics.
+            self.metrics.log_dict(
```

Comment:
Does `log_dict` compute a running mean or stores all returns?

Reply:
Depends on the leafs (which are Stats objects). If a leaf has reduce=mean, we mean, if it has reduce=sum, we sum, etc... Reduce can also be None (simply collect everything in a list). You can specify windows for the reduce operations or - by default - it'll use EMA with a coeff of 0.01 (you can change that as well).

It's super flexible and allows for:
* averaging over a window
* EMA averaging
* resetting upon reduce (good for per-iteration counts)
* summing without window (good for lifetime counts)
* max/min'ing with or without window
* delta timers and throughput measurements (which can also use the same reduction options as described above).